net/netcheck/netcheck.go

```diff
@@ -1175,11 +1177,12 @@ func (c *Client) measureHTTPSLatency(ctx context.Context, reg *tailcfg.DERPRegio
 	if err != nil {
 		return 0, ip, err
 	}
-	result.End(c.timeNow())
 
-	// TODO: decide best timing heuristic here.
-	// Maybe the server should return the tcpinfo_rtt?
-	return result.ServerProcessing, ip, nil
+	// return the connection duration, not the request duration, as this is the
```

Comment:
I'm generally in favor of simplifying this to TCP RTT, or TCP+TLS RTT if we can be sure the time is for one node, and not stretched across attempts to dial multiple in a given region. 

Something that has come up in a related PR is MITM proxies (also discussed up on L1169). One of the advantages of the previous TTFB-ish measurement is that it provides a measure that is comparable across regions, where TCP+TLS dial time may not be.

Reply:
oh, in my comment above, I probably misunderstood part of the issue, but with the data race that was going on before we had a similar problem where we'd have mixed data in the report sometimes.

Hurr.

I'm not sure I quite follow this part:

> One of the advantages of the previous TTFB-ish measurement is that it provides a measure that is comparable across regions, where TCP+TLS dial time may not be.

why is TTFB more comparable than connection RTT?

IIUC these times end up being compared to stun probes, so really we'd probably do well to move to a TCPing style probe for best comparison?

Reply:
considering:

client ---- MITM proxy ---- DERP

If MITM proxy is terminating TCP+TLS, TCP+TLS round trip will look similar across all regions, but `result.ServerProcessing` (TTFB) will surface relative differences of the leg between MITM proxy and DERP.

May not be an edge case worth considering.

Reply:
Oh yes, now I remember we discussed this before, thanks for reminding me again. It's a good point, also complex in the face of anywhere that the samples end up mixed up with STUN, but perhaps that's even more of an edge case. I'm imagining the common case of forced MITM is also probably either consistently blocking STUN or rarely blocking it rather than being mixed.

Ok, I think what I'll do here is that I'l actually ignore connection setup time. I'll instead measure the request "do time", the DERP endpoint shouldn't add substantial latency above TTFB, as it should be replying with the whole response header, if not the whole response in a single packet.